tests/e2e/specs/shopper/mini-cart-icon.test.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * Internal dependencies
+ */
+import { getBlockPagePermalink } from '../../../utils';
+
+const block = {
+	name: 'Mini Cart Block',
+};
+
+if ( process.env.WOOCOMMERCE_BLOCKS_PHASE < 3 )
+	// eslint-disable-next-line jest/no-focused-tests
+	test.only( `skipping ${ block.name } tests`, () => {} );
+
+describe( 'Shopper → Mini Cart → Can see the icon with a badge', () => {
+	it( 'Shopper can see the Mini Cart icon and it badge on the front end', async () => {
+		await page.goto( await getBlockPagePermalink( block.name ), {
+			waitUntil: 'networkidle0',
+		} );
```

Comment:
This approach is quite slow because we need to:
1. Search for the Mini Cart block page in the page list table.
2. Go to the edit page of the Mini Cart page.
3. Open the sidebar if needed.
4. Toggle the Permalink panel.
5. Get the link to the Mini Cart page.
6. Navigate to the page using `page.goto()`.

The bellow suggestion is an alternative way to navigate to the Mini Cart Block page.
Given that all the pages are added to the Storefront header navigation, we can navigate to the block page by clicking on the corresponding nav menu item. This sounds less reliable than the current approach but it's fine as long as we don't remove the menu items from the navigation.
```suggestion
		await page.goto( createURL( '/' ), { waitUntil: 'networkidle0' } );
		await expect( page ).toClick( '.nav-menu a', { text: block.name } );
		await page.waitForNavigation();
```

Reply:
Repost here for better visibility.

Reply:
This is a good idea, I'm thinking if in general since block pages creation is automated if we could in general - as we rely on that anyway - assume that certain page should exists and have a helper method that would goto page by block name/type following the same (possibly shared) algorithm for generating these pages. 

Did you run benchmarks by any chance as to how much extra time in E2E tests this costs us?

Reply:
>  Did you run benchmarks by any chance as to how much extra time in E2E tests this costs us?

Running `npm run test:e2e -- tests/e2e/specs/shopper/mini-cart-icon.test.js`:

With the current implementation: `PASS  tests/e2e/specs/shopper/mini-cart-icon.test.js (17.129 s)`

With the suggestion [above](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/5877#discussion_r810096719): `PASS  tests/e2e/specs/shopper/mini-cart-icon.test.js (11.913 s)`

Reply:
In [427648a](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/5877/commits/427648a43bdd1cc5ee68bea9c8a219ffe66166b6), I create a new utility for shoppers to visit a block page by its title. Because it's for shopper, I think it's more natural for a shopper to navigate to a page by the nav menu, so I apply the suggestion in this conversation for the utility instead of using `getBlockPagePermalink`.

Reply:
Nice, 35% faster - that adds up to a lot of carbon! 🏭 

And agree this will better reflect user's action. Thanks for setting it up.